src/device-state-access.h

```diff
@@ -139,11 +139,13 @@ class device_state_access : public i_device_state_access {
         return m_a.read_htif_iyield();
     }
 
-    // LCOV_EXCL_START
-    void do_write_memory(uint64_t paddr, const unsigned char *data, uint64_t log2_length) override {
-        return m_a.write_memory(paddr, data, log2_length);
+    bool do_read_memory(uint64_t paddr, unsigned char *data, uint64_t length) override {
```

Comment:
I am trying to understand what the implication of this is for uarch. What happens when you compile the interpreter to uarch? I see the method is not implemented there in uarch-machine-state-access.h. So the idea is that, if you are using virtio, you can't run the uarch? Or at least, it wouldn't work if you did? Should we add some test for this situation?

Reply:
So the interpreter that is for uarch, when it is executing a bit SD, and searching PMAs, doesn't find the PMA for the virtio devices?

Reply:
> So the interpreter that is for uarch, when it is executing a bit SD, and searching PMAs, doesn't find the PMA for the virtio devices?

Correct, VirtIO was not added to `build_uarch_pma_entry` in `uarch-machine-state-access.h`, SD will fail. But now I  added comments + `abort()` these places (`do_write_memory`, `do_read_memory`, `build_uarch_pma_entry`).

Plus I added a runtime error when trying to perform `uarch_run` with `iunrep` set, which should cover VirtIO being enabled.

Reply:
Great. I now feel safe. :)